emcc.py

```diff
@@ -533,23 +533,30 @@ def backend_binaryen_passes():
   # wasm backend output can benefit from the binaryen optimizer (in asm2wasm,
   # we run the optimizer during asm2wasm itself). use it, if not overridden.
 
+  # run the binaryen optimize in -O2+. in -O0 we don't need it obviously, while
+  # in -O1 we don't run it as the LLVM optimizer has been run, and it does the
+  # great majority of the work; not running the binaryen optimizer in that case
+  # keeps -O1 mostly-optimized while compiling quickly and without rewriting
+  # DWARF etc.
+  run_binaryen_optimizer = shared.Settings.OPT_LEVEL >= 2
+
   passes = []
   # safe heap must run before post-emscripten, so post-emscripten can apply the sbrk ptr
   if shared.Settings.SAFE_HEAP:
     passes += ['--safe-heap']
-  if shared.Settings.OPT_LEVEL > 0:
+  if run_binaryen_optimizer:
     passes += ['--post-emscripten']
     if not shared.Settings.EXIT_RUNTIME:
       passes += ['--no-exit-runtime']
-  if shared.Settings.OPT_LEVEL > 0 or shared.Settings.SHRINK_LEVEL > 0:
+  if run_binaryen_optimizer:
```

Comment:
you removed the check for SHRINK_LEVEL here but didn't add it back anywhere else. Should it go into `run_binaryen_optimizer`?

Reply:
This was intentional. `SHRINK_LEVEL` is for `-Os/-Oz` and in those modes we have `OPT_LEVEL >= 2` anyhow.